gcp_variant_transforms/testing/integration/run_tests.py

```diff
@@ -0,0 +1,254 @@
+    response = request.execute()
+    while not response['done']:
+      time.sleep(2)
+      response = request.execute()
+    return response
+
+
+class TestContextManager(object):
+  """Manages all resources for a given run of tests.
+
+  Responsible for setting up tests (i.e. creating a unique dataset) and
+  cleaning up all resources after tests have been completed.
+  """
+
+  def __init__(self, args):
+    self.staging_location = args.staging_location
+    self.temp_location = args.temp_location
+    self.logging_location = args.logging_location
+    self.project = args.project
+
+  def __enter__(self):
+    self.credentials = GoogleCredentials.get_application_default()
+    client = bigquery.Client(project=self.project)
+    self.dataset_id = 'integration_tests_{}'.format(
+        datetime.now().strftime('%Y%m%d_%H%M%S'))
+    dataset_ref = client.dataset(self.dataset_id)
+    self.dataset = bigquery.Dataset(dataset_ref)
+    _ = client.create_dataset(self.dataset)
+    return self
+
+  def __exit__(self, *args):
+    client = bigquery.Client(project=self.project)
+    tables = client.list_dataset_tables(self.dataset)
+    # Delete tables, otherwise dataset deletion will fail because it is still
+    # "in use".
+    for table in tables:
+      client.delete_table(table)
+    client.delete_dataset(self.dataset)
+
+
+def _get_args():
+  parser = argparse.ArgumentParser()
+  parser.add_argument('--project', required=True)
+  parser.add_argument('--staging_location', required=True)
+  parser.add_argument('--temp_location', required=True)
+  parser.add_argument('--logging_location', required=True)
+  return parser.parse_args()
+
+
+def _get_test_configs():
+  """Gets all test configs in integration directory and subdirectories."""
+  test_configs = []
+  test_file_path = os.path.join(
+      os.getcwd(), 'gcp_variant_transforms/testing/integration')
+  for root, _, files in os.walk(test_file_path):
+    for filename in files:
+      if filename.endswith('.json'):
+        test_configs.append(_load_test_config(os.path.join(root, filename)))
+  return test_configs
+
+
+def _load_test_config(filename):
+  """Loads an integration test JSON object from a file."""
+  with open(filename, 'r') as f:
+    test = json.loads(f.read())
+    _validate_test(test, filename)
+    return test
+
+
+def _validate_test(test, filename):
+  required_keys = ['test_name', 'table_name', 'input_pattern']
+  for key in required_keys:
+    if key not in test:
+      raise ValueError('Test case in {} is missing required key: {}'.format(
+          filename, key))
+
+
+def _create_tests(test_case_configs, context):
+  return [TestCase(context, **config) for config in test_case_configs]
+
+
+def _create_test(test_case_config, context):
+  """Creates a TestCase from test config."""
+  return TestCase(context, **test_case_config)
+
+
+def _run_test(test, context):
+  return test.run(context)
+
+
+def _handle_failure(response):
```

Comment:
can this be inside TestCase class as well?

Reply:
Done.